app/javascript/components/ansible-credentials-form/ansible-credentials-form.schema.js

```diff
@@ -0,0 +1,50 @@
+import { componentTypes, validatorTypes } from '@@ddf';
+
+const normalize = (fields, edit) => fields.map(({ component, label, helperText, fields, ...field }) => ({
+  ...field,
+  component,
+  label: __(label),
+  helperText: __(helperText),
+  ...(component === 'password-field' && { edit }),
+  ...(fields && { fields: normalize(fields, edit) }),
+}));
+
+const createSchema = (fields, promise, edit, loadSchema) => ({
+  fields: [
+    {
+      component: componentTypes.TEXT_FIELD,
+      id: 'name',
+      name: 'name',
+      label: __('Name'),
+      maxLength: 128,
+      validate: [{ type: validatorTypes.REQUIRED }],
+      isRequired: true,
+    },
+    {
+      component: componentTypes.SELECT,
+      id: 'type',
+      name: 'type',
+      label: __('Credential type'),
+      placeholder: __('<Choose>'),
+      validate: [{ type: validatorTypes.REQUIRED }],
+      isRequired: true,
+      isDisabled: edit,
+      onChange: value => promise.then(loadSchema(value)),
+      loadOptions: () =>
+        promise.then(
```

Comment:
You can move this line one line above and then spare one indentation level for the block below, prettier.io isn't always giving us the best results.